runtime/criusupport/criusupport.cpp

```diff
@@ -245,10 +250,43 @@ getNativeString(J9VMThread *currentThread, j9object_t javaString, char **nativeS
 }
 
 /**
- * Caller must first acquire exclusive safepoint or exclusive VMAccess
+ * Checks if a thread should be suspended or resumed for checkpoint/restore. If JDWP is not
+ * enabled, then all threads will return true. Otherwise, if toggleDebugThreads is true, then all
+ * JDWP debug threads will be toggled. If toggleDebugThreads is false, then all non-debug threads
+ * will be toggled.
+ *
+ * @param[in] currentThread the thread being checked
+ * @param[in] toggleDebugThreads if only JWDP debug threads should be toggled
+ *
+ * @return true if the thread should be suspended or resumed, false if not
+ */
+static bool
+shouldToggleJavaThread(J9VMThread *currentThread, BOOLEAN toggleDebugThreads)
+{
+	J9JavaVM *vm = currentThread->javaVM;
+	if (vm->checkpointState.isJdwpEnabled) {
+		char *threadName = getOMRVMThreadName(currentThread->omrVMThread);
+		releaseOMRVMThreadName(currentThread->omrVMThread);
+		if (toggleDebugThreads) {
+			/* all threads started by JDWP begin with "JDWP" in their name */
+			return 0 == strncmp("JDWP", threadName, 4);
+		} else {
+			return 0 != strncmp("JDWP", threadName, 4);
+		}
+	}
+	return true;
```

Comment:
I would suggest the following to avoid duplicated `strncmp` calls and have a single return point.
```
	J9JavaVM *vm = currentThread->javaVM;
	bool result = true;
	if (vm->checkpointState.isJdwpEnabled) {
		char *threadName = getOMRVMThreadName(currentThread->omrVMThread);
		releaseOMRVMThreadName(currentThread->omrVMThread);
		/* all threads started by JDWP begin with "JDWP" in their name */
		bool isJDWPThread = 0 == strncmp("JDWP", threadName, 4);
		if (toggleDebugThreads) {
			result = isJDWPThread;
		} else {
			result = !isJDWPThread;
		}
	}
	return result;
```